src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy

```diff
@@ -88,8 +88,9 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     }
 
     private static void writeToJcr(ProtoNode nodeProto, Session session) {
```

Comment:
All this logic needs to be moved within ProtoNodeDecorator, the above "void write()" will be : 

```groovy
void write(List<? extends ProtoNode> nodeProtos) throws Exception {
  Session session = theSession()
  for(ProtoNode node : nodeProtos) {
    new ProtoNodeDecorator(node).writeToJcr(session)
  }
}